kubernetes/cache/namespaces.go

```diff
@@ -18,14 +18,25 @@ type (
 func (c *kialiCacheImpl) SetNamespaces(token string, namespaces []models.Namespace) {
 	defer c.tokenLock.Unlock()
 	c.tokenLock.Lock()
-	nameNamespace := make(map[string]models.Namespace, len(namespaces))
+	nameNamespace := make(map[string]map[string]models.Namespace)
+	clusterNamespace := make(map[string]map[string]models.Namespace)
 	for _, ns := range namespaces {
-		nameNamespace[ns.Name] = ns
+		if nameNamespace[ns.Name] == nil {
+			nameNamespace[ns.Name] = make(map[string]models.Namespace)
+		}
+		nameNamespace[ns.Name][ns.Cluster] = ns
+
+		if clusterNamespace[ns.Cluster] == nil {
+			clusterNamespace[ns.Cluster] = make(map[string]models.Namespace)
+		}
+		clusterNamespace[ns.Cluster][ns.Name] = ns
 	}
+
 	c.tokenNamespaces[token] = namespaceCache{
-		created:       time.Now(),
-		namespaces:    namespaces,
-		nameNamespace: nameNamespace,
+		created:          time.Now(),
+		namespaces:       namespaces,
+		nameNamespace:    nameNamespace,
+		clusterNamespace: clusterNamespace,
```

Comment:
At the moment just the clusterNamespace is used in GetNamespace. 
Depending on how we request the information it would be more efficient have one structure or the other, but, at the moment, we can just leave the one that is used - or is more efficient.